source/slang/lookup.cpp

```diff
@@ -410,7 +410,27 @@ void lookUpMemberImpl(
     if (auto declRefType = type->As<DeclRefType>())
     {
         auto declRef = declRefType->declRef;
-        if (auto aggTypeDeclRef = declRef.As<AggTypeDecl>())
+        if (auto assocTypeDeclRef = declRef.As<AssocTypeDecl>())
+        {
+            for (auto constraintDeclRef : getMembersOfType<GenericTypeConstraintDecl>(assocTypeDeclRef))
+            {
+                // The super-type in the constraint (e.g., `Foo` in `T : Foo`)
+                // will tell us a type we should use for lookup.
```

Comment:
Not worth fixing right now, but I wonder if it is possible to share any of this logic between associated types and generic type parameters, since they both follow a pattern of enumerating their constraints and doing lookup through them...